p2p/src/main/kotlin/maru/p2p/MaruPeerManager.kt

```diff
@@ -68,6 +69,7 @@ class MaruPeerManager(
 
   fun stop(): SafeFuture<Unit> {
     stopCalled = true
+    scheduler.shutdown()
```

Comment:
I didn't add it here intentionally. We rely on the following rule: `start() can be called after stop()`, but `scheduler` can't be started back up again after `shutdown` call. So either we create the `scheduler` inside `start`, or we don't terminate it on `stop`

Reply:
Done